hlint-run has build failures with typescript 5.0

With typescript 5.0, `hlint-run` has build failures at these locations:
- https://github.com/haskell/actions/blob/5ad9f8c2edfe8b16e4c10fbe0c4767f6dd30bc5b/hlint-run/src/run.ts#L48
- https://github.com/haskell/actions/blob/5ad9f8c2edfe8b16e4c10fbe0c4767f6dd30bc5b/hlint-run/src/run.ts#L62

https://github.com/haskell/actions/actions/runs/4581184122/jobs/8090527193?pr=223#step:5:7
> Error: src/run.ts(48,39): error TS2365: Operator '>' cannot be applied to types 'string | number' and 'number'.
> Error: src/run.ts(62,41): error TS2365: Operator '>' cannot be applied to types 'string | number' and 'number'.

_Originally posted by @andreasabel in https://github.com/haskell/actions/issues/223#issuecomment-1492859183_

--- a/hlint-run/package.json
+++ b/hlint-run/package.json
@@ -33,6 +33,6 @@
   "devDependencies": {
     "@types/node": "^18.16.3",
     "@vercel/ncc": "^0.36.1",
-    "typescript": "^4.9.5"
+    "typescript": "^5.0.4"
   }
 }

--- a/hlint-run/src/run.ts
+++ b/hlint-run/src/run.ts
@@ -45,7 +45,7 @@ async function runHLint(cmd: string, args: string[]): Promise<HLintResult> {
 function getOverallCheckResult(failOn: CheckMode, {ideas, statusCode}: HLintResult): CheckResult {
   const hintsBySev = HLINT_SEV_LEVELS.map(sev => ([sev, ideas.filter(hint => hint.severity === sev).length]));
   const hintSummary = hintsBySev
-    .filter(([_sevName, numHints]) => numHints > 0)
+    .filter(([_sevName, numHints]) => +numHints > 0)
     .map(([sev, num]) => `${sev} (${num})`).join(', ');
 
   let ok: boolean;
@@ -59,7 +59,7 @@ function getOverallCheckResult(failOn: CheckMode, {ideas, statusCode}: HLintResu
     // Note that the summary still shows all counts.
     const failedBySev = hintsBySev
       .slice(0, HLINT_SEV_LEVELS.indexOf(failOn) + 1)
-      .filter(([_sevName, numHints]) => numHints > 0);
+      .filter(([_sevName, numHints]) => +numHints > 0);
     ok = failedBySev.length === 0;
   }
 

